lib/api/completeMultipartUpload.js

```diff
@@ -155,19 +164,59 @@ function completeMultipartUpload(authInfo, request, log, callback) {
             }
             return next(errors.MalformedXML, destBucket);
         },
-        function retrieveParts(destBucket, objMD, mpuBucket, jsonList, next) {
-            services.getMPUparts(mpuBucket.getName(), uploadId, log,
+        function getMPUMetadata(destBucket, objMD, mpuBucket, jsonList, next) {
+            return metadata.getObjectMD(mpuBucket.getName(), mpuOverviewKey,
+            {}, log, (err, storedMetadata) => {
+                if (err) {
+                    return next(err, destBucket);
+                }
+                const location = storedMetadata.controllingLocationConstraint;
+                return next(null, destBucket, objMD, mpuBucket, jsonList,
+                storedMetadata, location);
+            });
+        },
+        function ifMultipleBackend(destBucket, objMD, mpuBucket, jsonList,
+        storedMetadata, location, next) {
+            if (config.backends.data === 'multiple') {
+                return multipleBackendGateway.completeMPU(objectKey,
+                uploadId, location, jsonList, log, (err, newObjData) => {
+                    if (err) {
+                        return next(err, destBucket);
+                    }
+                    return next(null, destBucket, objMD, mpuBucket,
+                    jsonList, storedMetadata, newObjData);
+                });
+            }
+            return next(null, destBucket, objMD, mpuBucket, jsonList,
+            storedMetadata, null);
+        },
+        function retrieveParts(destBucket, objMD, mpuBucket, jsonList,
+        storedMetadata, newObjData, next) {
+            // means MPU completion was done on third-party backend,
+            // so part info is not in our metadata and related steps
+            // should be skipped
+            if (newObjData) {
+                return next(null, destBucket, objMD, mpuBucket, null,
+                jsonList, storedMetadata, newObjData);
+            }
+            return services.getMPUparts(mpuBucket.getName(), uploadId, log,
                 (err, result) => {
                     if (err) {
                         return next(err, destBucket);
                     }
                     const storedParts = result.Contents;
                     return next(null, destBucket, objMD, mpuBucket,
-                        storedParts, jsonList);
+                        storedParts, jsonList, storedMetadata, null);
                 });
         },
         function processParts(destBucket, objMD, mpuBucket, storedParts,
-            jsonList, next) {
+            jsonList, storedMetadata, newObjData, next) {
+            let calculatedSize = 0;
```

Comment:
the calculatedSize line was just moved there from further down in the function so I could return it in the if statement. Since it was just a one-line move, I thought it would be better to pass the variable than just 'null'